Python/ceval.c

```diff
@@ -387,13 +387,28 @@ Py_AddPendingCall(int (*func)(void *), void *arg)
     return result;
 }
 
+void
+_PyEval_SignalReceived(void)
+{
+    /* bpo-30703: Function called when the C signal handler of Python gets a
+       signal. We cannot queue a callback using Py_AddPendingCall() since this
+       function is not reentrant (use a lock and a list). */
```

Comment:
Also say "signal-safe" rather than "reentrant", since otherwise it's not clear how Py_AddPendingCall can be called reentrantly.